param/ipython.py

```diff
@@ -134,7 +134,14 @@ def _build_table(self, info, order, max_col_len=40, only_changed=False):
         (params, val_dict, changed) = info
         col_widths = dict((k,0) for k in order)
 
-        for name, p in params.items():
+        ordering = sorted(
+            sorted(params), # alphanumeric tie-breaker
+            key=lambda k: (- float('inf')  # No precedence is lowest possible precendence
+                           if params[k].precedence is None else
+                           params[k].precedence))
+
+        for name in ordering:
+            p = params[name]
```

Comment:
Seems like this would overwrite the definition order?  I think it should respect the definition order unless explicitly overridden with prececence. Probably the best bet is to move the sorting code from Panel (https://github.com/holoviz/panel/blob/master/panel/param.py#L506) into a function that can be called here and from Panel. First step is to copy it here, then make it general, then Panel can be updated later.